src/ILCompiler/src/Program.cs

```diff
@@ -342,6 +353,7 @@ private int Run(string[] args)
                         if (entrypointModule != null)
                             throw new Exception("Multiple EXE modules");
                         entrypointModule = module;
+                        r2rCompilerContext?.InitializeAlgorithm(module.MetadataReader.GetTableRowCount(TableIndex.TypeDef));
```

Comment:
Can this be done in some file within ILCompiler.ReadyToRun? In the compilation builder's `ToCompilation`, or constructor, or something? I would really like to avoid weird codegen/target specific quirks in this file. We have been doing pretty well on that front so far (even though this file already handles CppCodegen, WebAssembly and RyuJIT).